tests/testthat/test-clean-names.R

```diff
@@ -1,42 +1,197 @@
+    make_clean_names("average # of days"),
+    "average_number_of_days"
+  )
+  expect_equal(
+    make_clean_names("jan2009sales"),
+    "jan2009sales"
+  )
+  expect_equal(
+    make_clean_names("jan 2009 sales"),
+    "jan_2009_sales"
+  )
+  expect_equal(
+    make_clean_names("not_first_unicode_µ"),
+    "not_first_unicode_m"
+  )
+  expect_equal(
+    make_clean_names("µ_first_unicode"),
+    "m_first_unicode"
+  )
+  expect_equal(
+    make_clean_names("a/b"),
+    "a_b",
+    info="No custom replacement"
+  )
+  expect_equal(
+    make_clean_names("a/b", replace=c("/"="_per_")),
+    "a_per_b",
+    info="Custom replacement"
+  )
 })
 
+testing_vector <-
+  c(
+    "sp ace",
+    "repeated",
+    "a**^@",
+    "%",
+    "*",
+    "!",
+    "d(!)9",
+    "REPEATED",
+    "can\"'t",
+    "hi_`there`",
+    "  leading spaces",
+    "€",
+    "ação",
+    "Farœ",
+    "a b c d e f",
+    "testCamelCase",
+    "!leadingpunct",
+    "average # of days",
+    "jan2009sales",
+    "jan 2009 sales",
+    "not_first_unicode_µ",
+    "µ_first_unicode"
+  )
+result_vector_noascii <-
+  c(
+    "sp_ace", # spaces
+    "repeated", # first instance of repeat
+    "a", # multiple special chars, trailing special chars
+    "percent", # converting % to percent
+    "x", # 100% invalid name
+    "x_2", # repeat of invalid name
+    "d_9", # multiple special characters
+    "repeated_2", # uppercase, 2nd instance of repeat
+    "cant", # uppercase, 2nd instance of repeat
+    "hi_there", # double-underscores to single
+    "leading_spaces", # leading spaces
+    "x_3", # euro sign, invalid
+    "acao", # accented word, transliterated to latin,
+    "faroe", # Farœ character was failing to convert on Windows, should work universally for stringi 1.1.6 or higher
+    # https://github.com/sfirke/janitor/issues/120#issuecomment-303385418
+    "a_b_c_d_e_f", # for testing alternating cases below with e.g., case = "upper_lower"
+    "test_camel_case", # for testing alternating cases below with e.g., case = "upper_lower"
+    "leadingpunct", # for testing alternating cases below with e.g., case = "upper_lower"
+    "average_number_of_days", # for testing alternating cases below with e.g., case = "upper_lower"
+    "jan2009sales", # no separator around number-word boundary if not existing already
+    "jan_2009_sales", # yes separator around number-word boundary if it existed
+    "not_first_unicode_µ",
+    "µ_first_unicode"
+  )
+
+result_vector_ascii <-
+  c(
+    "sp_ace", # spaces
+    "repeated", # first instance of repeat
+    "a", # multiple special chars, trailing special chars
+    "percent", # converting % to percent
+    "x", # 100% invalid name
+    "x_2", # repeat of invalid name
+    "d_9", # multiple special characters
+    "repeated_2", # uppercase, 2nd instance of repeat
+    "cant", # uppercase, 2nd instance of repeat
+    "hi_there", # double-underscores to single
+    "leading_spaces", # leading spaces
+    "x", # euro sign, converted
```

Comment:
Why is this `x` and not `x_3` like in the no_ascii result vector?

More generally, is there test coverage showing the difference between `ascii = TRUE` and `FALSE`?  That was the only test that seemed missing.  For instance above `ação` turns into `acao` for both `TRUE` and `FALSE`.

Reply:
Not being `x_3` is part of the rework where I was having some issues with my testing.  I added a test for `ascii=TRUE`, but the number of ways that transliterations happen makes it difficult to fully control allow for e.g. the acao scenario.

I'm guessing that at the end of this, we'll have some additional work with Unicode characters and other non-ASCII characters, but I think that in the general cases, we're doing better.  (And more to the point, I don't see a way to do all that we are wanting to do without some crunching of unusual characters.)